test/dist_autograd_test.py

```diff
@@ -220,6 +223,28 @@ def test_rpc_complex_args(self):
             dst_rank = (self.rank + 1) % self.world_size
             self.assertEqual(worker_ids[0], dst_rank)
 
+
+    @dist_init
+    def test_context_cleanup_many_workers(self):
+        global known_context_ids
+        dst_ranks = {rank for rank in range(self.world_size) if rank != self.rank}
+        with dist_autograd.context() as context_id:
+            t1 = torch.ones(3, 3, requires_grad=True)
+            t2 = torch.zeros(3, 3, requires_grad=True)
+            for dst_rank in dst_ranks:
+                ret = rpc.rpc_sync("worker{}".format(dst_rank), torch.add, args=(t1, t2))
```

Comment:
Create a gh issue that we should add a test with multiple nested RPCs (with Python UDFs) once https://github.com/pytorch/pytorch/pull/27576 lands.